In [1]:
!python -m pip install aider-install
!aider-install

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 17.4/17.4 MB 34.1 MB/s eta 0:00:00
Resolved 114 packages in 3.35s
⠙ Preparing packages... (0/114)
⠙ Preparing packages... (0/114)
cffi       ------------------------------     0 B/468.19 KiB
⠙ Preparing packages... (0/114)
pydantic   ------------------------------     0 B/433.50 KiB
cffi       ------------------------------     0 B/468.19 KiB
⠙ Preparing packages... (0/114)
prompt-toolkit ------------------------------     0 B/378.72 KiB
pydantic   ------------------------------     0 B/433.50 KiB
cffi       ------------------------------     0 B/468.19 KiB
⠙ Preparing packages... (0/114)
yarl       ------------------------------     0 B/341.03 KiB
prompt-toolkit ------------------------------     0 B/378.72 KiB
pydantic   ------------------------------     0 B/433.50 KiB
cffi       ------------------------------     0 B/468.19 KiB
⠙ Preparing packages... (0/114)
yarl       ------------------------------     0 B/341.03 KiB
prompt-toolkit ----

In [1]:
!export GEMINI_API_KEY=XXXXXXXXXXXXXXXXXXXXX

In [4]:
!export PATH="/root/.local/bin:$PATH"

In [2]:
!export PATH="$PATH:/root/.local/bin"

In [3]:
!aider --model gemini

/bin/bash: line 1: aider: command not found


In [4]:
!which aider

In [ ]:
!export PATH="$PATH:/path/to/aider"

In [2]:
!export PATH="$PATH:/root/.local/bin/aider"

In [5]:
!export PATH="$PATH:/root/.local/bin/"

In [6]:
!echo $PATH

/opt/bin:/usr/local/nvidia/bin:/usr/local/cuda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/tools/node/bin:/tools/google-cloud-sdk/bin


In [7]:
!aider --model gemini

/bin/bash: line 1: aider: command not found


In [8]:
!python -m aider


/usr/bin/python3: No module named aider


In [9]:
!pip install colab-xterm

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 115.6/115.6 kB 2.3 MB/s eta 0:00:00


In [10]:
%load_ext colabxterm

In [ ]:
%xterm


In [ ]:
!pip install colab-xterm
%load_ext colabxterm

%xterm
curl -fsSL https://ollama.com/install.sh | sh

In [ ]:
python -m pip install aider-install
aider-install
export GEMINI_API_KEY=<key>
aider --model gemini-exp


Requirement already satisfied: uv>=0.5.0 in /usr/local/lib/python3.11/dist-packages (from aider-install) (0.7.3)
Resolved 114 packages in 1.07s
Prepared 1 package in 1ms
Uninstalled 1 package in 9ms
Installed 1 package in 10ms
 ~ aider-chat==0.82.3
Installed 1 executable: aider
Executable directory /root/.local/bin is already in PATH
/content# export GEMINI_API_KEY=XXXXXXXXXXXXX
/content# aider --model gemini-exp
aider --model gemini-exp
──────────────────────────────────────────────────────────────────────────────────────
No git repo found, create one to track aider's changes (recommended)? (Y)es/(N)o [Yes]
:                                                                                     
Added .aider* to .gitignore
Git repository created in /content
Update git name with: git config user.name "Your Name"
Update git email with: git config user.email "you@example.com"
Aider v0.82.3
Main model: gemini/gemini-2.5-pro-exp-03-25 with diff-fenced edit format
Weak model: gemini/gemini-2.5-flash-preview-04-17
Git repo: .git with 0 files
Repo-map: using 4096 tokens, auto refresh


https://aider.chat/HISTORY.html#release-notes
Would you like to see what's new in this version? (Y)es/(N)o [Yes]:                   
Please answer with one of: yes, no, skip, all
Would you like to see what's new in this version? (Y)es/(N)o [Yes]:                   
──────────────────────────────────────────────────────────────────────────────────────
> hi                                                                                  

Hi there! How can I help you with your code today?